ostd/src/arch/x86/trap/mod.rs

```diff
@@ -220,25 +223,37 @@ pub fn is_kernel_interrupted() -> bool {
 /// Handle traps (only from kernel).
 #[no_mangle]
 extern "sysv64" fn trap_handler(f: &mut TrapFrame) {
+    // Ensures that the IRQ state during the trap that handles exceptions remains
+    // consistent with the state before entering.
```

Comment:
```suggestion
    // The IRQ state before trapping. We need to ensure that the IRQ state
    // during exception handling is consistent with the state before the trap.
```

This should start with a sentence describing the `was_irq_enabled` flag.